server/src/main/java/com/flipkart/varadhi/web/routes/RouteBehaviour.java

```diff
@@ -7,7 +7,9 @@
 @Getter
 public enum RouteBehaviour {
     authenticated(100),
-    hasBody(200);
+    hasBody(200),
```

Comment:
well that ordering (if such things are being enforced), is vertx concern.
The routeBehaviour's user is not / should not be concerned with such things. The code which is setting up httpServer should be the one to configure the handlers according to vertx.

Reply:
So from the source code, this is how the weight is calculated: 
```
private static Priority weight(Handler<RoutingContext> handler) {
    if (handler instanceof PlatformHandler) {
      return Priority.PLATFORM;
    }
    if (handler instanceof SecurityPolicyHandler) {
      return Priority.SECURITY_POLICY;
    }
    if (handler instanceof ProtocolUpgradeHandler) {
      return Priority.PROTOCOL_UPGRADE;
    }
    if (handler instanceof BodyHandler) {
      return Priority.BODY;
    }
    if (handler instanceof MultiTenantHandler) {
      return Priority.MULTI_TENANT;
    }
    if (handler instanceof AuthenticationHandler) {
      return Priority.AUTHENTICATION;
    }
    if (handler instanceof InputTrustHandler) {
      return Priority.INPUT_TRUST;
    }
    if (handler instanceof AuthorizationHandler) {
      return Priority.AUTHORIZATION;
    }

    return Priority.USER;
  }
```

The issue occurs when I tried enabling the authentication config. The map of route behaviour to configurator orders keys based on the enum `order` field. In that it causes the body handler to be added last which causes the above issue. Fix is to set the order value for hasBody to be less than the others.

Reply:
Since we are using vertx web BodyHandler this ordering is being enforced.

Reply:
for custom handlers it is assigned USER priority which is the lowest

Reply:
I agree with us taking the approach of our own w.r.to handlers chaining instead of Vertx enforcement. This would require some additional work w.r.to adding custom implementation of such vertx provided implementation. For time being we can move body handler as first one in the chain to unblock and bring it back by providing the custom implementation.